cpp/src/arrow/array/array_nested.cc

```diff
@@ -680,7 +680,7 @@ Result<std::shared_ptr<Array>> SparseUnionArray::GetFlattenedField(
   if (child_null_bitmap) {
     BitmapAnd(flattened_null_bitmap->data(), child_offset, child_null_bitmap->data(),
               child_offset, child_data->length, child_offset,
-              flattened_null_bitmap->mutable_data());
+              flattened_null_bitmap->mutable_data(), /*out pop count*/ nullptr);
```

Comment:
Make`nullptr` the default value to this new argument to leave current code untouched if they don't care about pop count.

Reply:
I did think about that, but wasn't sure if default arguments are used in exported functions, especially in the functions that had pointers and basic types compared to bitmap functions that returned class objects.

Is there any C API in arrow where this would apply ? such as API's that are wrapped around from other languages like Rust ?

Reply:
Good questions. AFAIK, arrow c++ is called by arrow c/python/r implementation. Not sure if an exported c++ function with default parameter is problematic.
@lidavidm @kou

Reply:
It should not be a problem. One, we make use of default parameters quite extensively already, and if we need a binding, we can handle it appropriately (if needed with an "extern C" though I don't think we need resort to that). Two, this function is mostly used internally and is unlikely to be used by a binding (and it's in the ::internal namespace anyways).

Reply:
I'm not sure that the problem case correctly but I think that there are no problem with default parameter.
(Is this ABI compatibility problem?)
Because we increase major version for each release.